### DISCUSSION TITLE
Wide, index-like TOC

### DISCUSSION THREAD

**Discussion by travelster** [allejo/jekyll-toc#62] [Q&A] [answered]
How could a table of contents like the following mockup be automatically created?  The left side really isn't required, just the right side.  Being expanded like this provides the user a quick view of all it's contents.  

![Expanded TOC](https://user-images.githubusercontent.com/87431528/134007980-49cc7f7f-0982-4095-a129-b627c07a79e7.jpg)

**Comment by allejo** [comment, marked as answer]
Sure, this is entirely possible. If you have all of those articles in an array, let's say, `posts`, you would be able to loop through all of them and just feed each post's content into `toc.html`. Then use the `base_url` parameter to point to another page (the HTML ID will automatically be appended).

```liquid
{% for post in posts %}
    {% assign postUrl = post.url | relative_url %}

    {% include toc.html html=post.content base_url=postUrl %}
{% endfor %}
```

^ that snippet is untested, but that's the idea of how you'd go about generating a site-wide TOC.
